codalab_service.py

```diff
@@ -518,7 +524,26 @@ def __init__(self, args):
             self.args.version = clean_version(self.args.version)
         self.compose_cwd = os.path.join(BASE_DIR, 'docker', 'compose_files')
 
-        self.compose_files = ['docker-compose.yml']
+        self.compose_files = []
+        self.compose_tempfile_name = ""
+        if self.args.link_mounts:
+            # We want to be able to mount a variable number of folders to the Docker container,
+            # so we can't just use regular interpolation with environment variables. Instead,
+            # we create a temporary file with the modified docker-compose.yml and use that file instead.
+            with open(os.path.join(self.compose_cwd, 'docker-compose.yml')) as f:
+                compose_options = yaml.load(f)
+            for mount_path in self.args.link_mounts.split(","):
+                mount_path = os.path.abspath(mount_path)
+                compose_options["x-codalab-server"]["volumes"].append(
+                    f"{mount_path}:/opt/codalab-worksheets-link-mounts{mount_path}"
+                )
+            with tempfile.NamedTemporaryFile(mode='w', suffix=".yml", delete=False) as f:
```

Comment:
Can we put this in `var_path` and name it something like `docker-compose-custom.yml`, so that it's easier to find / debug?

Reply:
Sounds good -- hopefully this won't cause issues with people concurrently running codalab_service.py, right? (as long as they specify different instance names, it should be fine, I guess)